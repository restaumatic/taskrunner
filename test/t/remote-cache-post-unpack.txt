# no toplevel
# s3

export TASKRUNNER_SAVE_REMOTE_CACHE=1

mkdir a

(
  cd a
  echo foo > input.txt
  git init -q
  git add input.txt
)

cp -r a b

go() {
  # Note: we want separate workdir for separate repos
  TASKRUNNER_STATE_DIRECTORY="$(pwd)" taskrunner -n mytask bash -e -c '
    snapshot input.txt --outputs output.txt --cmd "
      echo post-unpack stdout
      echo post-unpack stderr >&2
      cp output.txt output2.txt
    "
    echo "Expensive computation"
    rev input.txt > output.txt
  '
}

(
  cd a
  go
  echo "Output in a is: $(cat output.txt)"
  echo "Output2 in a is: $(cat output2.txt)"
)

(
  cd b
  go # should not run expensive computation, download instead, but should run post-unpack cmd
  echo "Output in b is: $(cat output.txt)"
  echo "Output2 in b is: $(cat output2.txt)"
)
