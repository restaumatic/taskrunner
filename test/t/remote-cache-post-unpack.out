-- output:
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation
[mytask] info | success
[mytask] stderr | post-unpack stdout
[mytask] stderr | post-unpack stderr
Output in a is: oof
Output2 in a is: oof
[mytask] info | Found remote cache mytask-6b36308659163b577fd1c832107dc46ca3aa659b.tar.zst, restoring
[mytask] info | Restored from remote cache
[mytask] stderr | post-unpack stdout
[mytask] stderr | post-unpack stderr
Output in b is: oof
Output2 in b is: oof
